provider/kubernetes/kubernetes.go

```diff
@@ -736,3 +791,58 @@ func getRateLimit(i *extensionsv1beta1.Ingress) *types.RateLimit {
 
 	return rateLimit
 }
+
+func createEmptyBackend(name string, templateObjects types.Configuration) (types.Configuration, error) {
+	if _, exists := templateObjects.Backends[name]; !exists {
```

Comment:
If you turn around the logic we could save one level of indent :) The comment below would also become obsolete as the error message documents the if then. The same holds true for the functions below.